go-update/src/test/java/org/reactome/release/goupdate/JavaVsPerlComparison.java

```diff
@@ -51,138 +54,173 @@ public static void main(String[] args) throws Exception
-							if (numDiffs > 0)
-							{
-								regulationDiffCount ++;
-								mainSB.append("\n***\nFor Java-updated referring Regulation instance \""+javaReferringRegulation.toString()+"\" there are " + numDiffs + " differences:\n"+sb1.toString());
-							}
-							else
-							{
-								regulationSameCount ++;
-							}
+							diffCount.incrementAndGet();
+							mainSB.append("\n***\nFor Java-updated instance \""+javaGoInst.toString()+"\" with accession \""+accession+"\" there are " + i + " differences:\n"+sb.toString());
 						}
-					}
-					else if (javaGoInst.getSchemClass().getName().equals(ReactomeJavaConstants.GO_CellularComponent))
-					{
-						StringBuilder sb1 = new StringBuilder();
-						@SuppressWarnings("unchecked")
-						List<GKInstance> javaList = ((List<GKInstance>) javaGoInst.getReferers(ReactomeJavaConstants.goCellularComponent));
-						@SuppressWarnings("unchecked")
-						List<GKInstance> perlList = ((List<GKInstance>) perlInst.getReferers(ReactomeJavaConstants.goCellularComponent));
-
-						if (javaList!=null && perlList!=null)
+						else
 						{
-							javaList = javaList.stream().sorted(dbIdComparator).collect(Collectors.toList());
-							perlList = perlList.stream().sorted(dbIdComparator).collect(Collectors.toList());
-							GKInstance javaReferringPE = javaList.get(0);
-							GKInstance perlReferringPE = perlList.get(0);
-
-							int numDiffs = DBObjectComparer.compareInstances(javaReferringPE, perlReferringPE, sb1, 0, predicate);
-							if (numDiffs > 0)
-							{
-								peDiffCount ++;
-								mainSB.append("\n***\nFor Java-updated referring PhysicalEntity instance \""+javaReferringPE.toString()+"\" there are " + numDiffs + " differences:\n"+sb1.toString());
-							}
-							else
+							sameCount.incrementAndGet();
+						}
+						Predicate<? super SchemaAttribute> isNameAttribute = p-> {
+								return p.getName().equals("name") || p.getName().equals("_displayName");
+							};
+						
+						// Now, we need to compare referrers, since they might have display name changes. Want to make sure we capture those correctly.
+						if (javaGoInst.getSchemClass().getName().equals(ReactomeJavaConstants.GO_BiologicalProcess))
+						{
+							StringBuilder sb1 = new StringBuilder();
+	
+							@SuppressWarnings("unchecked")
+							List<GKInstance> javaList = ((List<GKInstance>) javaGoInst.getReferers(ReactomeJavaConstants.goBiologicalProcess));
+							@SuppressWarnings("unchecked")
+							List<GKInstance> perlList = ((List<GKInstance>) perlInst.getReferers(ReactomeJavaConstants.goBiologicalProcess));
+							if (javaList != null && perlList != null)
 							{
-								peSameCount ++;
+								javaList = javaList.stream().sorted(dbIdComparator).collect(Collectors.toList());
+								perlList = perlList.stream().sorted(dbIdComparator).collect(Collectors.toList());
+	
+								GKInstance javaReferringRegulation =  javaList.get(0);
+								GKInstance perlReferringRegulation = perlList.get(0);
```

Comment:
The referrers actually could be a regulation or an event (pathway or reaction like event).  Usually it is an event, so it may be clearer to change the name.